src/policy/policy.cc

```diff
@@ -0,0 +1,155 @@
+#include "policy.h"
+#include "base_object-inl.h"
+#include "env-inl.h"
+#include "memory_tracker-inl.h"
+#include "node.h"
+#include "node_external_reference.h"
+#include "node_errors.h"
+
+#include "v8.h"
+
+#include <string>
+#include <vector>
+#include <iostream>
+
+namespace node {
+
+using v8::Array;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Integer;
+using v8::Just;
+using v8::Local;
+using v8::Maybe;
+using v8::Nothing;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+namespace policy {
+
+// The root policy is establish at process start using
+// the --policy-deny-* command line arguments.
+Policy root_policy;
+
+namespace {
+
+// policy.deny('fs.in', ['/tmp/'])
+// policy.deny('fs.in')
+static void Deny(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  v8::Isolate* isolate = env->isolate();
+
+  CHECK(args[0]->IsString());
+  CHECK(args.Length() >= 2 || args[1]->IsArray());
+
+  std::string denyScope = *String::Utf8Value(isolate, args[0]);
+  Permission scope = Policy::StringToPermission(denyScope);
+  if (scope == Permission::kPermissionsRoot) {
+    return args.GetReturnValue().Set(false);
+  }
+
+  Local<Array> jsParams = Local<Array>::Cast(args[1]);
+  std::vector<std::string> params;
+  for (uint32_t i = 0; i < jsParams->Length(); ++i) {
+    Local<Value> arg(
+        jsParams
+          ->Get(isolate->GetCurrentContext(), Integer::New(isolate, i))
+          .ToLocalChecked());
+
+    String::Utf8Value utf8_arg(isolate, arg);
+    params.push_back(*utf8_arg);
+  }
+
+  return args.GetReturnValue()
+    .Set(root_policy.Deny(scope, params));
```

Comment:
This should modify a per-Environment policy, not a global one. (It could modify a global one but in that case it needs to be protected against concurrent access; however, semantically speaking, this is a per-Environment value.) You may want to look at how https://github.com/nodejs/node/pull/22112 implemented this.